Dockerfile.j2

```diff
@@ -1,21 +1,30 @@
-FROM ubuntu:18.04 as amd64
-RUN apt update && apt install -y qemu-user-static
-
+# Built with arch: {{ arch }} flavor: {{ flavor }} image: {{ image }}
+#
 ################################################################################
 # base system
 ################################################################################
-FROM arm32v7/ubuntu:18.04 as system
-
+{%if arch == "amd64"%}
+FROM {{image}} as system
+{%elif arch == "armhf"%}
+# qemu helper for arm build
+FROM {{image}} as amd64
+RUN apt update && apt install -y qemu-user-static
+FROM arm32v7/{{image}} as system
 COPY --from=amd64 /usr/bin/qemu-arm-static /usr/bin/
-ARG localbuild
-RUN if [ "x$localbuild" != "x" ]; then sed -i 's#http://archive.ubuntu.com/#http://tw.archive.ubuntu.com/#' /etc/apt/sources.list; fi
+{%endif%}
 
+{% if localbuild %}
+RUN echo "*** LOCAL BUILD ***"
```

Comment:
One unused layer for final Docker image. If we squash it into existing RUN command, it will be bette. How do you think?

Reply:
Yes I've also spotted that one :)

https://github.com/jwilder/docker-squash could also be run from the Makefile.